homeassistant/components/utility_meter/select.py

```diff
@@ -82,12 +86,15 @@ class TariffSelect(SelectEntity, RestoreEntity):
     def __init__(
         self,
         name,
-        tariffs,
-        unique_id,
+        suggested_entity_id,
+        tariffs: list[str],
+        *,
+        unique_id: str | None = None,
         device_info: DeviceInfo | None = None,
     ) -> None:
         """Initialize a tariff selector."""
         self._attr_name = name
+        self.entity_id = suggested_entity_id
```

Comment:
I am a little confused, the PR is talking about setting a friendly name, but in reality we only set the entity_id to a fixed value. Isn't the solution to set `_attr_has_entity_name = True`?

Reply:
this line is to avoid a breaking change. 

The most important line in this PR is line 67:

` conf_meter_name = hass.data[DATA_UTILITY][meter].get(CONF_NAME, meter)`

which enables exchanging `meter` (the slug in yaml) for the name configured in YAML

The issue was that this simple change was breaking the entity_id, so I had to include the suggest_entity_id (much like it happens in utility_meter sensor platform)

Reply:
Does every utility meter have a unique_id?

Reply:
Config Entry yes, YAML no... (you need to explicitly set it)

Reply:
Because if you have a unique_id, the entity_id will stay the same. So I am mostly wondering if changes to names count as breaking change when it comes to YAML based integrations (I haven't had those yet and a lot of YAML stuff was before my time)

Reply:
If you have an old yaml configuration without unique_id it will break. It broke the tests 😅

Before unique_id I would use the yaml slug to assure uniqueness of utility meter and name the entity accordingly, so yes, this is very old legacy...

Reply:
> Before unique_id I would use the yaml slug to assure uniqueness of utility meter and name the entity accordingly, so yes, this is very old legacy...

Yes but I mean, if you had a YAML based entity and you changed the name in the code, was that considered a breaking change?

I personally don't like force overwriting the `entity_id`, since with this PR we are now introducing our own naming scheme and if the normal naming scheme changes (for whatever reason) the select entity would look weird again.

Reply:
There was no concept of breaking change at the time. 

The only way around here is to actually introduce the breaking change suggesting users to add the unique_id to their yaml configuration.

Reply:
I've narrowed down the backward compatibility

Reply:
I think it's fine to fix the name but the rest seems as not something that should be part of this PR.
I don't see why the user can't provide the unique id (as also is done in all other helpers configured from yaml).

Reply:
@gjohansson-ST it's legacy... it's from a time the slug in YAML was the unique_id

If you are concerned with lines 97-100 I can drop those

Reply:
Let's do that

Reply:
@gjohansson-ST done